Pracs/prac_07/project_management.py

```diff
@@ -115,7 +157,12 @@ def add_project(projects):
 
 def filter_projects(projects):
     """Filter projects by user input."""
-    pass
+    filter_date_string = input("Show projects that start after date (dd/mm/yy): ")
+    filter_date = datetime.datetime.strptime(filter_date_string, "%d/%m/%Y").date()
+    sorted_projects = sorted(projects, key=Project.sort_by_date)
+    for project in sorted_projects:
```

Comment:
Rather than sorting the whole list, you could use list comprehension to create a smaller filtered list based on the requested date and reduce needing to write another method in your class:
`filtered_projects = [project for project in projects if project.start_date >= filter_date]`
You could then sort this list by the date and then print each project.

Reply:
Ah yes that makes sense, thanks!